### PR TITLE
chore(can): socket can reusability

### PR DESCRIPTION
**Pull request by amitlissack**
Extracting the socketcan code to its own include directory.  The classes are now templates.

### CODE REVIEW

File: can/CMakeLists.txt [hunk 1]
```diff
@@ -5,6 +5,7 @@ else()
     if("${CMAKE_HOST_SYSTEM_NAME}" STREQUAL "Linux")
         # Simulator requires linux only kernel interfaces
         add_subdirectory(simulator)
```
Comment (Laura-Danielle) [inline can/CMakeLists.txt:7]: Would it mess things up to make the simulator the static library here rather than creating a separate directory?

Comment (amitlissack) [reply can/CMakeLists.txt]: I decided to not make a shared library. The reason is that the socket can code depended on FreeRTOS which is a pain using shared libraries due to the freertosconfig.h requirement.

I removed the dependency and made the socket can stuff templates.

File: can/simulator/main.cpp [hunk 2]
```diff
@@ -5,9 +5,10 @@
 #include "can/core/freertos_can_dispatch.hpp"
 #include "can/core/message_writer.hpp"
 #include "can/core/messages.hpp"
-#include "can/simulator/sim_canbus.hpp"
-#include "can/simulator/socket_can.hpp"
+#include "can/simlib//sim_canbus.hpp"
```
Comment (sfoster1) [inline can/simulator/main.cpp:8]:
```suggestion
#include "can/simlib/sim_canbus.hpp"
```

File: include/common/core/freertos_synchronization.hpp [hunk 3]
```diff
@@ -61,4 +61,19 @@ class FreeRTOSMutexFromISR {
...
+  public:
+    FreeRTOSCriticalSection() {}
+    FreeRTOSCriticalSection(const FreeRTOSCriticalSection &) = delete;
+    FreeRTOSCriticalSection(const FreeRTOSCriticalSection &&) = delete;
+    FreeRTOSCriticalSection &operator=(const FreeRTOSCriticalSection &) =
+        delete;
+    FreeRTOSCriticalSection &&operator=(const FreeRTOSCriticalSection &&) =
+        delete;
+
+    void acquire() { taskENTER_CRITICAL(); }
```
Comment (Laura-Danielle) [inline include/common/core/freertos_synchronization.hpp:74]: Are we purposefully trying to disable interrupts as well here?

**Review by Laura-Danielle — APPROVED** [review]
Looks good, just had one question about the freertos lock.

File: include/common/core/freertos_synchronization.hpp (hunk 3 above)
Comment (amitlissack) [reply include/common/core/freertos_synchronization.hpp:74]: In this case, yes. The socketcan transport classes uses a blocking read and needs to not have it be interrupted by FreeRTOS.

I don't know of a use case for this class on the firmware.

**Pull request merged**